src/services/completions.ts

```diff
@@ -3053,6 +3049,18 @@ namespace ts.Completions {
         }
     }
 
+    function isApparentType(type: Type, node: ObjectLiteralExpression | JsxAttributes, checker: TypeChecker) {

```

Comment:
What does 'apparent' mean here? I think this needs a different name because of interference from the checker term 'apparent'.

In the checker, an apparent type is (1) the constraint of a type parameter (2) the wrapper type of a primitive type (eg `String` is the apparent type of `string`) or (3) just the type itself, for any other type.